### ISSUE TITLE
Upgrade to version 2 spec

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/x86-64-assembly-test-runner#19]
We've recently updated the [test runner interface specification](https://github.com/exercism/docs/blob/main/building/tooling/test-runners/interface.md) to allow for two types of test runners, identified via a new `version` property in the `results.json` file (see [the spec for the version property](https://github.com/exercism/docs/blob/main/building/tooling/test-runners/interface.md#version)):

1. `version: 1` test runners are quite basic, and only detect if either all tests passed or not. These test runners capture the console output of the test runner and output that. They do _not_ include details on individual tests passing/failing
2. `version: 2` test runners contain detailed information on individual tests passing/failing. We've recently added a required field named `test_code`, which contains the code the test ran to verify the behavior (see [the spec for the test_code property](https://github.com/exercism/docs/blob/main/building/tooling/test-runners/interface.md#test-code)).

The Java test runner's output is currently what I refer to as version 1.5: it contains the individual test information, but does not yet contain the `test_code` key. If possible, the Java test runner should be updated to also include the test code, in which case the `version` property should also be added with `2` as its value.

You are completely free on how you want to extract the test code, either via the AST or via finding the test code in the source code text via string manipulation. 

Let me know if there are any questions.

**Comment by hut8** [comment]
The version key is present now, so is that why tests.sh fails? I'm trying to test the runner locally and it fails for seemingly this reason.

**Comment by ErikSchierboom** [comment]
@hut8 Yep. I've just opened https://github.com/exercism/x86-64-assembly-test-runner/pull/34 to fix this.

**Comment by hut8** [comment]
@ErikSchierboom you're awesome. Thanks for your work and quick reply. 👍🏻

**Comment by ErikSchierboom** [comment]
@hut8 In case you hadn't seen, the PR was merged so this should be fixed now.

**Comment by hut8** [comment]
👍🏻 Saw it. This can be closed then, right?

**Comment by ErikSchierboom** [comment]
Not entirely, as this bit has not been implemented:

> If possible, the test runner should be updated to also include the test code
